core/trino-main/src/main/java/io/trino/type/DoubleOperators.java

```diff
@@ -55,7 +57,7 @@ public final class DoubleOperators
     private static final double MIN_BYTE_AS_DOUBLE = -0x1p7;
     private static final double MAX_BYTE_PLUS_ONE_AS_DOUBLE = 0x1p7;
 
-    private static final ThreadLocal<DecimalFormat> FORMAT = ThreadLocal.withInitial(() -> new DecimalFormat("0.0###################E0"));
```

Comment:
Can we get rid of the threadlocal here @findepi ?

Reply:
I understand the: 
```
 * Decimal formats are generally not synchronized.
 * It is recommended to create separate format instances for each thread.
 * If multiple threads access a format concurrently, it must be synchronized
 * externally.
```

but can we solve it with some other pattern than ThreadLocal?

Reply:
what would you suggest as an alternative?

Reply:
Some `ObjectPool` implementation?

Reply:
would it be better, or just different?
unnecessary synchronization?

Reply:
I'd say that ThreadLocals needs to go away and replaced with ScopedValue when this API finally lands.

Reply:
i don't think ScopedValue is applicable here.

anyway, feels orthogonal to the change offered in this PR